src/components/TextInput/TextInput.jsx

```diff
@@ -33,11 +41,12 @@ const TextInput = ({
         success && styles.success,
         wrapperClassNames,
         !label && styles.withoutLabel
-      )}>
+      )}> 
       <input
         id={id}
         ref={input}
         placeholder={placeholder || " "}
+        size={isPlaceholder(placeholder) ? placeholder.length : undefined}
         className={classNames(styles.textinput, inputClassNames)}
         type={type}
         {...props}
```

Comment:
We spread the props object here so people can customize TextInputs outside the pi-ui package. That means you can just pass a `size` prop from Decrediton and pi-ui will just "forward" it to the HTML input.